cli/src/pcluster/models/cluster.py

```diff
@@ -281,8 +281,8 @@ def __init__(
     ):
         super().__init__()
         self.assign_public_ip = Param(assign_public_ip)
-        self.security_groups = security_groups
-        self.additional_security_groups = additional_security_groups
+        self.security_groups = Param(security_groups)
```

Comment:
security_groups and additional_security_groups are List[str] type. Instead of wrapping them to Param, we could wrap the element in list to Param. Sometimes, elements in list is a combination of default values and customer specified values. 
In BaseSchema, `remove_implied_values ` will iterate the list and remove implied values.

But if all the elements are default values or customer specified values, just ignore my comment 😄

Reply:
TBH I'm not a big fan of this `Param` class in general. Anyway I think that a configuration property is either entirely computed or not (a "partial" computation does not seem to me to make sense). I mean, even if some items in the list are implied and others not, it's always the entire list that must be computed, for instance to remove duplicates. 
Does it make sense?

Reply:
Yes. We need more discussion on if we want to use `Param` or not. There are several questions to be answered:
1. Do we want to keep `implied` for each field?
2. Besides using `Param`, is there a better way to keep the `implied`?
3. If `Param` has to be used, where and how should we use it?

Reply:
I just want to confirm that the Param represent a parameter in the configuration file, and in this case the param is AdditionalSecurityGroups or SecurityGroups. It's a single Param that contains a list of string and not a list of Param.

Instead we can discuss about the Param usage. We can consider to insert a map in the Resource to store all the information related to the Params instead of attaching them to the param itself.
For now let's go ahead with this Param approach that works as expected altough not the optimal one, then we can review it. Thanks for all the feedbacks!